src/pages/library/DataExplorer.js

```diff
@@ -1,30 +1,107 @@
-import { commonPaths } from 'src/components/breadcrumbs'
-import { Fragment } from 'react'
-import { div, h } from 'react-hyperscript-helpers'
+import _ from 'lodash/fp'
+import { Component, Fragment } from 'react'
+import { div, h, span } from 'react-hyperscript-helpers'
 import IframeResizer from 'react-iframe-resizer-super'
-import TopBar from 'src/components/TopBar'
+import { ajaxCaller } from 'src/libs/ajax'
+import { authStore, contactUsActive } from 'src/libs/state'
+import { centeredSpinner } from 'src/components/icons'
+import { commonPaths } from 'src/components/breadcrumbs'
+import { link } from 'src/components/common'
 import * as Nav from 'src/libs/nav'
 import * as Style from 'src/libs/style'
+import TopBar from 'src/components/TopBar'
+import * as Utils from 'src/libs/utils'
+
+// name must be name from Data Explorer dataset.json
+// authDomain must be authorization_domain from Data Explorer dataset.json
+const datasets = [
+  {
+    name: '1000 Genomes',
+    origin: 'https://test-data-explorer.appspot.com'
+  },
+  {
+    name: 'AMP PD - 2019_v1beta_0220',
+    origin: 'https://amp-pd-data-explorer.appspot.com',
+    authDomain: 'amp-pd-researchers'
+  },
+  {
+    name: 'Baseline Health Study',
+    origin: 'https://baseline-baseline-explorer.appspot.com',
+    authDomain: 'baseline-researchers'
+  },
+  {
+    name: 'Nurses\' Health Study',
+    origin: 'https://nhs-explorer.appspot.com',
+    authDomain: 'nhs_saturn_users'
+  },
+  {
+    name: 'UK Biobank',
+    origin: 'https://biobank-explorer.appspot.com',
+    authDomain: 'Kathiresan_UKBB'
+  }
+]
 
+const DataExplorer = _.flow(
+  ajaxCaller,
+  Utils.connectAtom(authStore, 'authState')
+)(class DataExplorer extends Component {
+  constructor(props) {
+    super(props)
+    const { dataset } = props
+    this.state = {
+      completedDeOauth: undefined,
+      groups: undefined
+    }
+    this.origin = datasets.filter(d => d.name === dataset)[0].origin
+    this.authDomain = datasets.filter(d => d.name === dataset)[0].authDomain
+  }
 
-const datasetToUrl = {
-  // key must be dataset name from Data Explorer dataset.json
-  '1000 Genomes': 'https://test-data-explorer.appspot.com/?embed'
-}
+  async componentDidMount() {
+    const { ajax: { Groups } } = this.props
+    const groups = _.map(g => g.groupName, await Groups.list())
```

Comment:
This call is failing on the 1000 Genomes page if you're logged out

Reply:
Thanks. Let me know if I need to check registrationStatus/acceptedTos/profile in addition to `isSignedIn`.

I also skipped `fetch(favicon.ico)` if user is logged out, so you don't see the fetch error in console.